frontend/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -30,58 +30,87 @@ export interface PlotlyChartProps {
   height: number | undefined
 }
 
+export interface PlotlyIFrameProps {
+  width: number
+  height: number | undefined
+  url: string
+}
+
 export const DEFAULT_HEIGHT = 450
 
-export function PlotlyChart({
-  width: propWidth,
-  element,
+function renderIFrame({
+  url,
+  width,
   height: propHeight,
+}: PlotlyIFrameProps): ReactElement {
+  const height = propHeight || DEFAULT_HEIGHT
+  return <iframe title="Plotly" src={url} style={{ width, height }} />
+}
+
+function renderFigure({
+  element,
+  width,
+  height,
 }: PlotlyChartProps): ReactElement {
-  const renderIFrame = (url: string): ReactElement => {
-    const height = propHeight || DEFAULT_HEIGHT
-    const width = propWidth
-    return <iframe title="Plotly" src={url} style={{ width, height }} />
-  }
+  const figure = element.figure as FigureProto
+  const isFullScreen = (): boolean => !!height
 
-  const isFullScreen = (): boolean => !!propHeight
+  const [savedElement, saveElement] = useState(element)
+  const [config, setConfig] = useState(JSON.parse(figure.config))
+  const theme: Theme = useTheme()
+  const [savedTheme, saveTheme] = useState(theme)
 
   const generateSpec = (figure: FigureProto): any => {
     const spec = JSON.parse(figure.spec)
 
     if (isFullScreen()) {
-      spec.layout.width = propWidth
-      spec.layout.height = propHeight
+      spec.layout.width = width
+      spec.layout.height = height
     } else if (element.useContainerWidth) {
-      spec.layout.width = propWidth
+      spec.layout.width = width
     }
 
-    const theme: Theme = useTheme()
     spec.layout = layoutWithThemeDefaults(spec.layout, theme)
 
     return spec
   }
-
-  const renderFigure = (figure: FigureProto): ReactElement => {
-    const config = JSON.parse(figure.config)
-    const { data, layout, frames } = generateSpec(figure)
-
-    return (
-      <Plot
-        key={isFullScreen() ? "fullscreen" : "original"}
-        className="stPlotlyChart"
-        data={data}
-        layout={layout}
-        config={config}
-        frames={frames}
-      />
-    )
+  const [spec, setSpec] = useState(generateSpec(figure))
+
+  const elementChanged = !(savedElement && savedElement === element)
```

Comment:
Wouldn't it also be possible here to use a conditional effect (or layout effect)? Thereby, we might not need to actually save the element and theme in the state.

For example:

```typescript
useEffect(() => {
    setConfig(JSON.parse(figure.config))
    setSpec(generateSpec(figure))
}, [element, theme]);
```